builder/openstack/step_run_source_server.go

```diff
@@ -57,6 +58,7 @@ func (s *StepRunSourceServer) Run(state multistep.StateBag) multistep.StepAction
 		CreateOptsBuilder: servers.CreateOpts{
 			Name:             s.Name,
 			ImageRef:         s.SourceImage,
+			ImageName:        s.SourceImageName,
```

Comment:
You are right, it is not defined in OpenStack Docs, but gophercloud impelements such option and does translation from name to ID https://github.com/rackspace/gophercloud/blob/master/openstack/compute/v2/servers/requests.go#L270